drivers/char/broadcom/vc_mem.c

```diff
@@ -214,11 +173,34 @@ vc_mem_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 	return rc;
 }
 
-/****************************************************************************
-*
-*   vc_mem_mmap
-*
-***************************************************************************/
+#ifdef CONFIG_COMPAT
+static long
+vc_mem_compat_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
+{
+	int rc = 0;
+
+	switch (cmd) {
+	case VC_MEM_IOC_MEM_PHYS_ADDR32:
+		pr_debug("%s: VC_MEM_IOC_MEM_PHYS_ADDR32=0x%p\n",
+			 __func__, (void *)mm_vc_mem_phys_addr);
+
+		/* This isn't correct, but will cover us for now as
+		 * VideoCore is 32bit only.
+		 */
+		if (copy_to_user((void *)arg, &mm_vc_mem_phys_addr,
+				 sizeof(compat_ulong_t)) != 0)
```

Comment:
Missed one (sorry)

Reply:
:-(

Reply:
Cheer up - the rest looked OK.

Reply:
Updated.

I may regret trying to clean up 64 bit builds - the number of instances of %d/%x for size_t, or slightly dodgy pointer arithmetic, is depressing.

Reply:
But their all just 32-bit ints, aren't they? ;-)